docs/lang/articles/advanced/meta.md

```diff
@@ -4,92 +4,91 @@ sidebar_position: 1
 
 # Metaprogramming
 
-Taichi provides metaprogramming infrastructures. Metaprogramming can
+Taichi provides metaprogramming infrastructures. There are many benefits of metaprogramming in Taichi:
 
-- Unify the development of dimensionality-dependent code, such as
-  2D/3D physical simulations
-- Improve run-time performance by from run-time costs to compile time
-- Simplify the development of Taichi standard library
+- Enabling the development of dimensionality-independent codes, e.g., codes which are
+  adaptive for both 2D/3D physical simulations.
+- Improving runtime performance by moving computations from runtime to compile time.
+- Simplifying the development of Taichi standard library.
 
-Taichi kernels are _lazily instantiated_ and a lot of computation can
-happen at _compile-time_. Every kernel in Taichi is a template kernel,
-even if it has no template arguments.
+:::note Taichi kernels are **lazily instantiated** and massive computations can
+be executed at **compile-time**. Every kernel in Taichi is a template kernel,
+even if it has no template arguments. :::
 
 ## Template metaprogramming
 
-You may use `ti.template()` as a type hint to pass a field as an
-argument. For example:
+By using `ti.template()` as a argument type hint, a Taichi field can be passed into a kernel. Template programming also enable the codes to be reused for fields with different shape:
 
 ```python {2}
 @ti.kernel
-def copy(x: ti.template(), y: ti.template()):
+def copy1D(x: ti.template(), y: ti.template()):
```

Comment:
Shall we use snake case such as `copy_1D` or `copy_1d`?

Reply:
Have modified the function names to the snake case.